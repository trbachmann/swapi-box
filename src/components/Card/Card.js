import React from 'react';

export const Card = ({item, category}) => {
  let card;
  switch(item.type) {
    case 'people':
      card = <div>
        <h3>{item.name}</h3>
        <p>Homeworld: {item.homeworld_name}</p>
        <p>Species: {item.species_name}</p>
        <p>Population: {item.population}</p>
        <button>Add to Favorites</button>
      </div>;
      break;
    case 'planets':
      card = <div>
        <h3>Name</h3>
        <p>Terrain: </p>
        <p>Population: </p>
        <p>Climate:</p>
        <p>Residents: </p>
        <button>Add to Favorites</button>
      </div>;
      break;
    case 'vehicles':
      card = <div>
        <h3>{item.name}</h3>
        <p>Model: {item.model}</p>
        <p>Class: {item.vehicle_class}</p>
        <p>Number of Passengers: {item.passengers}</p>
        <button>Add to Favorites</button>
      </div>;
      break;
    default: 
      card = <h2>Sorry there are no cards for your selection. Please choose a different category.</h2>;
  }
  return(
    <div className='card'>
      {card}
    </div>
  )
}
