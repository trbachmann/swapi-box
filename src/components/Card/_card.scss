.card {
  border-radius: 15px;
  background: rgba(196, 196, 196, 0.31);
  height: 350px;

  .inner-card {
  margin: 1.2em;
  }
  
  .card-title {
    display: flex;
    justify-content: space-between;
    align-items: center;
  }

  h3 {
    color: $yellow;
  }

  p {
    color: white;
  }

  button {
    color: $yellow;
    font-size: 1em;
    margin: auto;
    border: solid $yellow 2px;
  }

  ul {
    color: white;
    height: 2.4em;
    overflow-y: scroll;
  }

  .favorite-false {
    color: white;
    margin-left: auto;
    &:hover {
      color: $yellow;
    }
  }

    .favorite-true {
    color: $yellow;
    margin-left: auto;
    &:hover {
      color: white;
    }
  }
}




