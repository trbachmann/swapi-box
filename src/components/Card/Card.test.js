import React from 'react';
import { Card } from './Card';
import { shallow } from 'enzyme';

const mockPerson = { type: 'people', name: 'Han', homeworld_name: 'Alderaan', species_name: 'human', population: '200000', created: '239D'}
const mockPlanet = { type: 'planets', name: 'Hoth', terrain: 'rocky', population: '3200000', climate: 'temperate', residents: ['R2D2', 'Mark'], created:'275F' };
const mockVehicle = { type: 'vehicles', name: 'Crawler', model: 'Digger Crawler', class: 'construction', passengers: '1', created: '2987C'};
const mockHandleFavorite = jest.fn();
const mockIsFavorite = true;
describe('Card', () => {
  
  it('should match the snapshot with people data passed in', () => {
    const wrapper = shallow(<Card 
      item={mockPerson} 
      handleFavorite={mockHandleFavorite}
      isFavorite={mockIsFavorite}
    />);
    expect(wrapper).toMatchSnapshot();
  });

  it('should match the snapshot with planet data passed in', () => {
    const wrapper = shallow(<Card
      item={mockPlanet}
      handleFavorite={mockHandleFavorite}
      isFavorite={mockIsFavorite}
    />);
    expect(wrapper).toMatchSnapshot();
  });

  it('should match the snapshot with people data passed in', () => {
    const wrapper = shallow(<Card
      item={mockVehicle}
      handleFavorite={mockHandleFavorite}
      isFavorite={mockIsFavorite}
    />);
    expect(wrapper).toMatchSnapshot();
  });

  it('should call handleFavorite once when a user clicks the fa-star icon', () => {
    const wrapper = shallow(<Card
      item={mockPerson}
      handleFavorite={mockHandleFavorite}
      isFavorite={mockIsFavorite}
    />);

    wrapper.find('i.fa-star').simulate('click');
    expect(mockHandleFavorite).toHaveBeenCalledTimes(1);
  });

  it('should call handleFavorite with the item.created prop when a user clicks the fa-star icon', () => {
    const wrapper = shallow(<Card
      item={mockPerson}
      handleFavorite={mockHandleFavorite}
      isFavorite={mockIsFavorite}
    />);
    wrapper.find('i.fa-star').simulate('click');
    expect(mockHandleFavorite).toHaveBeenCalledWith(mockPerson.created);
  })
});