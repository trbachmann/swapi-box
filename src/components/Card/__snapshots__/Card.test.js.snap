// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Card should match the snapshot with people data passed in 1`] = `
<div
  className="card"
>
  <div
    className="inner-card"
  >
    <div
      className="card-title"
    >
      <h3>
        Han
      </h3>
      <i
        className="fas fa-star favorite-true"
        onClick={[Function]}
      />
    </div>
    <p>
      Homeworld: 
      Alderaan
    </p>
    <p>
      Species: 
      human
    </p>
    <p>
      Population: 
      200000
    </p>
  </div>
</div>
`;

exports[`Card should match the snapshot with people data passed in 2`] = `
<div
  className="card"
>
  <div
    className="inner-card"
  >
    <div
      className="card-title"
    >
      <h3>
        Crawler
      </h3>
      <i
        className="fas fa-star favorite-true"
        onClick={[Function]}
      />
    </div>
    <p>
      Model: 
      Digger Crawler
    </p>
    <p>
      Class: 
    </p>
    <p>
      Number of Passengers: 
      1
    </p>
  </div>
</div>
`;

exports[`Card should match the snapshot with planet data passed in 1`] = `
<div
  className="card"
>
  <div
    className="inner-card"
  >
    <div
      className="card-title"
    >
      <h3>
        Hoth
      </h3>
      <i
        className="fas fa-star favorite-true"
        onClick={[Function]}
      />
    </div>
    <p>
      Terrain: 
      rocky
    </p>
    <p>
      Population: 
      3200000
    </p>
    <p>
      Climate: 
      temperate
    </p>
    <p>
      Residents: 
    </p>
    <ul>
      <li
        key="R2D2"
      >
        R2D2
      </li>
      <li
        key="Mark"
      >
        Mark
      </li>
    </ul>
  </div>
</div>
`;
